Update PR #28417, add era modifier to TauRefProducer to read old data format from existing files

swozniewski opened the pull request:
#### PR description:

Addressing issue #29448 
Given workflows read root files with old tauID format which requires an era modifier to adapt the configuration accordingly which is done in this PR.

#### PR validation:

Ran 136.7801 successfully

Commit:
add process modifier to TauRefProducer to read old data format from existing files

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-29467/14663

- This PR adds an extra 20KB to repository

cmsbuild commented:
A new Pull Request was created by  @swozniewski  for master.

It involves the following packages:

Configuration/Eras
DQMOffline/Trigger


@andrius-k, @kmaeshima, @schneiml, @cmsbuild, @silviodonato, @franzoni, @jfernan2, @fioriNTU, @fabiocos, @davidlange6 can you please review it and eventually sign? Thanks.
@missirol, @makortel, @mtosi, @Fedespring, @Martin-Grunewald, @calderona, @HuguesBrun, @jhgoh, @cericeci, @trocino, @fabiocos, @folguera, @rociovilar this is something you requested to watch as well.
@silviodonato, @dpiparo you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

slava77 commented on Configuration/Eras/python/Era_Run2_2017_cff.py:
```diff
@@ -22,8 +22,9 @@
 Run2_2017 = cms.ModifierChain(Run2_2016.copyAndExclude([run2_muon_2016, run2_HLTconditions_2016, run2_ECAL_2016, run2_egamma_2016,pixel_2016,run2_jme_2016, strips_vfp30_2016]),
                               phase1Pixel, run2_ECAL_2017, run2_HF_2017, run2_HCAL_2017, run2_HE_2017, run2_HEPlan1_2017, 
-                              trackingPhase1, run2_GEM_2017, stage2L1Trigger_2017, run2_HLTconditions_2017, run2_muon_2017,run2_egamma_2017, ctpps_2017, run2_jme_2017)
+                              trackingPhase1, run2_GEM_2017, stage2L1Trigger_2017, run2_HLTconditions_2017, run2_muon_2017,run2_egamma_2017, ctpps_2017, run2_jme_2017, run2_tauIDformat)
```
I think that this modifier should be added to the workflow, not to the era.
IIUC, this current update will break other workflows that actually rerun tau reco.

swozniewski replied:
Ok, could you point me to the configuration of these workflows?

slava77 replied:
I think that the step definition is here https://github.com/cms-sw/cmssw/blob/959f56649b489b3c33a15bd502bef3df4f31ea78/Configuration/PyReleaseValidation/python/relval_steps.py#L2671-L2679

swozniewski replied:
Thanks! I'll implement it as process modifier then

slava77 commented on Configuration/Eras/python/Modifier_run2_tauIDformat_cff.py:
```diff
@@ -0,0 +1,4 @@
+import FWCore.ParameterSet.Config as cms
+
+run2_tauIDformat =  cms.Modifier()
```
why does this have run2_ prefix? IIRC, #28417 added tau container-based ID for all data/simulation eras/ranges.

I was actually expecting that the modifier will add the sequence/task to rerun tau IDs rather than insert support to read the old IDs, but for this case it seems like your solution is less invasive.

I propose to change the name of the modifier to ```tau_readOldDiscriminator ``` or something similar

swozniewski replied:
My reasoning was that files produced during run2 contain the old format (which could be called run2_tauIDformat), but I'm fine with calling it `tau_readOldDiscriminator` which better reflects what it is doing. However, one might think that this means to switch to a different ID rather than just an old format. Would `tau_readOldDiscriminatorFormat` also suitable or is it too long?

slava77 replied:
"during run2" is just a calendar period, there are AOD files produced for Run-1, Run-3 and Phase-2 during that period. It would be rather confusing to use run2_ prefix for a modifier for these cases as well if it becomes necessary.

swozniewski replied:
true

jfernan2 commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 22362fc929775108a7292f41d3091a674dfde8e3
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-61ce3b/5679/summary.html
CMSSW: CMSSW_11_1_X_2020-04-13-1100
SCRAM_ARCH: slc7_amd64_gcc820

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
The code-checks are being triggered in jenkins.

swozniewski commented:
@slava77 I switched to a process modifier and added it via the `--procModifiers` option to the workflow you pointed at. However, I don't see this option in the printout of the cmsDriver command when running the 136.7801 workflow. (scram has been run) Do you have an idea what is missing?

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-29467/14665

- This PR adds an extra 52KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Configuration/PyReleaseValidation/python/relval_steps.py  modified in PR(s): #29457

cmsbuild commented:
Pull request #29467 was updated. @pgunnell, @andrius-k, @silviodonato, @chayanit, @wajidalikhan, @schneiml, @kpedro88, @cmsbuild, @kmaeshima, @franzoni, @jfernan2, @fioriNTU, @fabiocos, @davidlange6 can you please check and sign again.

slava77 commented on Configuration/PyReleaseValidation/python/relval_steps.py:
```diff
@@ -2690,6 +2690,7 @@ def gen2021HiMix(fragment,howMuch):
     '--secondfilein':'filelist:step1_dasparentquery.log',
     '--customise_commands':'"process.HLTAnalyzerEndpath.remove(process.hltL1TGlobalSummary)"',
     '--fileout':'DQMHLTonAOD.root',
+    '--procModifiers':'tau_readOldDiscriminatorFormat',
```
IIUC, the problem shows up in DQMHLTonAOD_2017 as well (used in 136.7801).
This will address only 136.7803

slava77 commented:
> @slava77 I switched to a process modifier and added it via the `--procModifiers` option to the workflow you pointed at. However, I don't see this option in the printout of the cmsDriver command when running the 136.7801 workflow. (scram has been run) Do you have an idea what is missing?

I commented inline; it looks like you edited a line for a different workflow

swozniewski commented:
Ah I see. I thought, they share this part. Will quickly test and update

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-29467/14666

- This PR adds an extra 52KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Configuration/PyReleaseValidation/python/relval_steps.py  modified in PR(s): #29457

cmsbuild commented:
Pull request #29467 was updated. @pgunnell, @andrius-k, @silviodonato, @chayanit, @wajidalikhan, @schneiml, @kpedro88, @cmsbuild, @kmaeshima, @franzoni, @jfernan2, @fioriNTU, @fabiocos, @davidlange6 can you please check and sign again.

slava77 commented:
@cmsbuild please test 136.7801,136.7803

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-61ce3b/5679/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 34
 - DQMHistoTests: Total histograms compared: 2695166
 - DQMHistoTests: Total failures: 299
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2694548
 - DQMHistoTests: Total skipped: 319
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 33 files compared)
 - Checked 147 log files, 16 edm output root files, 34 DQM output files

slava77 commented:
@cmsbuild please test workflow 136.7801,136.7803

cmsbuild commented:
The tests are being triggered in jenkins.
Test Parameters:
  - **MATRIX_EXTRAS** = 136.7801,136.7803

cmsbuild commented:
+1
Tested at: 87de900f4edd555b515bd7fc2210a872de98cfae
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-61ce3b/5683/summary.html
CMSSW: CMSSW_11_1_X_2020-04-13-1100
SCRAM_ARCH: slc7_amd64_gcc820

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-61ce3b/5683/summary.html

@slava77 comparisons for the following workflows were not done due to missing matrix map:
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-61ce3b/136.7801_RunHLTPhy2017B_AOD+RunHLTPhy2017B_AOD+DQMHLTonAOD_2017+HARVESTDQMHLTonAOD_2017
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-61ce3b/136.7803_RunHLTPhy2017B_RAWAOD+RunHLTPhy2017B_RAWAOD+DQMHLTonRAWAOD_2017+HARVESTDQMHLTonAOD_2017

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 34
 - DQMHistoTests: Total histograms compared: 2695166
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2694846
 - DQMHistoTests: Total skipped: 319
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 33 files compared)
 - Checked 147 log files, 16 edm output root files, 34 DQM output files

silviodonato commented:
urgent

silviodonato commented:
+operations

jfernan2 commented:
@swozniewski The change in DQMOffline/Trigger/python/HLTTauDQMOffline_cfi.py is producing a  lost in the entries for the denominator taken to calculate efficiencies in HLT / TAU / PFTaus / folder
Is this understood? Thanks

slava77 commented:
> @swozniewski The change in DQMOffline/Trigger/python/HLTTauDQMOffline_cfi.py is producing a lost in the entries for the denominator taken to calculate efficiencies in HLT / TAU / PFTaus / folder
> Is this understood? Thanks

In the comparisons: DQMHistoTests: Total failures: 1
Did you look at the older/buggy diff?

swozniewski commented:
Does your question refer to the workflows fixed in this PR or to the original PR? Because here, we use the same inputs now as before and there should be no difference. If the new data format is used, there may be small changes due to the usage of ByLooseCombinedIsolationDBSumPtCorr3Hits because its configuration was unified with other cut based IDs. But the change should only be very small. So it depends on how large the loss is that you observe.

kpedro88 commented:
+upgrade

jfernan2 commented:
+dqm
Sorry, it seems I looked to the comparison from 18h ago, not the last one 14h ago. Apologies

chayanit commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @silviodonato, @dpiparo (and backports should be raised in the release meeting by the corresponding L2)

silviodonato commented:
+1
